packages/buckets/src/api/index.ts

```diff
@@ -631,8 +531,114 @@ export async function bucketsPushPathNode(
     stream.end()
   })
 }
+
+/**
+ * Pushes an iterable of files to a bucket.
+ * @param key Unique (IPNS compatible) identifier key for a bucket.
+ * @param input The input array of file/stream/objects.
+ * @param opts Options to control response stream.
+ * @internal
+ */
+export function bucketsPushPaths(
+  api: GrpcConnection,
+  key: string,
+  input: any,
+  opts?: Omit<PushOptions, 'progress'>,
+  ctx?: ContextInterface,
+): AsyncIterableIterator<PushPathsResult> {
+  return new Repeater<PushPathsResult>(async (push, stop) => {
+    const clientjs = new APIServiceClient(api.serviceHost, api.rpcOptions)
+
+    const metadata = { ...api.context.toJSON(), ...ctx?.toJSON() }
+
+    const stream: BidirectionalStream<
+      PushPathsRequest,
+      PushPathsResponse
+    > = clientjs.pushPaths(metadata)
+
+    if (opts?.signal !== undefined) {
+      opts.signal.addEventListener('abort', () => {
+        stream.cancel()
+        throw AbortError
+      })
+    }
+
+    stream.on('data', (message: PushPathsResponse) => {
+      // Let's just make sure we haven't aborted this outside this function
+      if (opts?.signal?.aborted) {
+        stream.cancel()
+        return stop(AbortError)
+      }
+      const obj: PushPathsResult = {
+        path: message.getPath(),
+        root: fromPbRootObjectNullable(message.getRoot()),
+        cid: new CID(message.getCid()),
+        pinned: message.getPinned(),
+        size: message.getSize(),
+      }
+      push(obj)
```

Comment:
Just push the object onto the iterable for consumers. We don't need a progress callback, because they can simply respond to progress within the async iterable loop.